src/libcrun/container.c

```diff
@@ -1046,26 +1145,34 @@ libcrun_configure_libkrun (struct container_entrypoint_s *args, libcrun_error_t
 }
 
 static int
-libcrun_configure_wasmer (struct container_entrypoint_s *args, libcrun_error_t *err)
+libcrun_configure_wasm (struct container_entrypoint_s *args, libcrun_error_t *err)
 {
-
-#if HAVE_DLOPEN && HAVE_WASMER
+#if HAVE_DLOPEN && (HAVE_WASMER || HAVE_WASMEDGE)
   void *handle;
 #endif
 
 #if HAVE_DLOPEN && HAVE_WASMER
   handle = dlopen ("libwasmer.so", RTLD_NOW);
-  if (handle == NULL)
-    return crun_make_error (err, 0, "could not load `libwasmer.so`: %s", dlerror ());
+  if (handle != NULL)
+    {
+      args->exec_func = wasmer_do_exec;
+      args->exec_func_arg = handle;
+      return 0;
+    }
+#endif
 
-  args->exec_func = wasmer_do_exec;
-  args->exec_func_arg = handle;
+#if HAVE_DLOPEN && HAVE_WASMEDGE
+  handle = dlopen ("libwasmedge_c.so", RTLD_NOW);
```

Comment:
will we leak `handle` if it was already created as part of `handle = dlopen ("libwasmer.so", RTLD_NOW);`?  I think we should guard the entire new block with if `(handle == NULL)`

What should happen when both `HAVE_WASMER` and `HAVE_WASMEDGE` are defined?

Reply:
no sorry, nevermind my comment, the leak doesn't apply because we have a `return 0;` in the previous block